lib/firestore/firestore.dart

```diff
@@ -82,6 +82,35 @@ class Firestore {
 
   DocumentReference document(String path) => DocumentReference(_gateway, path);
 
+  /// Executes the given [TransactionHandler] and then attempts to commit the
+  /// changes applied within an atomic transaction.
+  ///
+  /// In the [TransactionHandler], a set of reads and writes can be performed
+  /// atomically using the [Transaction] object passed to the [TransactionHandler].
+  /// After the [TransactionHandler] is run, [Firestore] will attempt to apply the
+  /// changes to the server. If any of the data read has been modified outside
+  /// of this [Transaction] since being read, then the transaction will be
+  /// retried by executing the provided [TransactionHandler] again. If the transaction still
+  /// fails after the specified [maxAttempts] retries, then the transaction will fail.
+  ///
+  /// The [TransactionHandler] may be executed multiple times, it should be able
+  /// to handle multiple executions.
+  ///
+  /// Data accessed with the transaction will not reflect local changes that
+  /// have not been committed. For this reason, it is required that all
+  /// reads are performed before any writes. Transactions must be performed
+  /// while online. Otherwise, reads will fail, and the final commit will fail.
```

Comment:
To avoid confusion with the term ["online" in algorithms](https://en.wikipedia.org/wiki/Online_algorithm).

```suggestion
  /// with an internet connection. Otherwise, reads will fail, and the final commit will fail.
```